module-system/module-implementations/sov-evm/src/query.rs

```diff
@@ -68,15 +63,69 @@ impl<C: sov_modules_api::Context> Evm<C> {
         evm_transaction.map(|tx| tx.into())
     }
 
+    // TODO https://github.com/Sovereign-Labs/sovereign-sdk/issues/502
     #[rpc_method(name = "getTransactionReceipt")]
     pub fn get_transaction_receipt(
         &self,
-        _hash: H256,
-        _working_set: &mut WorkingSet<C::Storage>,
+        hash: H256,
+        working_set: &mut WorkingSet<C::Storage>,
     ) -> Option<TransactionReceipt> {
-        unimplemented!("eth_getTransactionReceipt not implemented")
+        info!("evm module: eth_getTransactionReceipt");
+
+        let receipt = self.receipts.get(&hash.into(), working_set);
```

Comment:
(Non-blocking): Do we want receipts to live in State, or just be (optionally) available via RPC. I'm wondering if we should add another DB interface for modules which stores/retrieves information in native mode and is a no-op in zk mode.

Reply:
Yes, we will have to think of a better way of doing this, I will work on the design this week.